pahfit/model.py

```diff
@@ -387,30 +399,49 @@ def fit(
         # check if observed spectrum is compatible with instrument model
         instrument.check_range([min(x), max(x)], inst)
 
-        # weigths
-        w = 1.0 / uncz
-
-        # construct model and perform fit
-        astropy_model = self._construct_astropy_model(inst, z, use_instrument_fwhm)
-        fit = LevMarLSQFitter(calc_uncertainties=True)
-        self.astropy_result = fit(
-            astropy_model,
-            xz,
-            yz,
-            weights=w,
-            maxiter=maxiter,
-            epsilon=1e-10,
-            acc=1e-10,
+        self.fitter = self._set_up_fitter(
+            inst, z, x=x, use_instrument_fwhm=use_instrument_fwhm
         )
-        self.fit_info = fit.fit_info
+        self.fitter.fit(xz, yz, uncz, maxiter=maxiter)
+
+        # copy the fit results to the features table
+        self._ingest_fit_result_to_features(self.fitter)
+
         if verbose:
-            print(fit.fit_info["message"])
+            print(self.fitter.message)
 
-        self._parse_astropy_result(self.astropy_result)
+    def _ingest_fit_result_to_features(self, fitter: Fitter):
+        """Copy the results from a Fitter to the features table
 
-    def info(self):
-        """Print out the last fit results."""
-        print(self.astropy_result)
+        This is a utility method, executed only at the end of fit(),
+        where the Fitter is passed after Fitter.fit() has been applied.
+        Passing a different Fitter object could be useful for
+        testing.
```

Comment:
How would it be helpful to use a different `Fitter` than the one you just ran `fit` on?

Reply:
I think this is because you've been thinking in terms of building a new `Fitter` each time.  For testing I'd say make a new `Model` with a different `Fitter`, and pre-seed it with the `Features` table from the old model.

Reply:
I used this in a test, but I changed the way things work now. Instead of returning / taking a Fitter object, the internal utility functions of `Model` now just set and edit `self.fitter`